src/methods/validate-fixture-input.ts

```diff
@@ -27,6 +27,19 @@ export interface ValidateFixtureInputResult {
   errors: string[];
 }
 
+/**
+ * Tracks expected fields at a selection set level, distinguishing between:
+ * - common fields (selected outside inline fragments)
+ * - type-specific fields (selected inside inline fragments)
+ */
+interface ExpectedFields {
+  common: Set<string>; // Fields that should be present on all objects
+  byType: Map<
+    GraphQLNamedType,
+    { fields: Set<string>; possibleTypes: Set<string> }
+  >; // Fields specific to inline fragment types
```

Comment:
I found this a bit hard to reason through why `GraphQLNamedType` was they key on the map. I would've expected it to be a `string` for the object type, and then the value be a `Set<string>` with all the fields for that object type. I think doing it this way would make `checkForExtraFields` simpler.